### ISSUE TITLE
animation for chain extention

### ISSUE THREAD

**Issue by Francesco215** [Francesco215/requirements_for_self-organization_video#13]
we need an animation that adds element to the chain of elements below

<img width="825" alt="Screenshot 2023-11-27 alle 18 36 46" src="https://github.com/Francesco215/requirements_for_self-organization_video/assets/47751420/49adfdc8-9212-4c06-bb34-5b66115d4d33">

and in the end becomes like this
<img width="817" alt="Screenshot 2023-11-27 alle 18 36 14" src="https://github.com/Francesco215/requirements_for_self-organization_video/assets/47751420/cefef23e-c2e7-441b-be76-471c868c89d9">

since at the start the shorter chain already occupies almost the entire screen, as you draw all the other circles it will need to un-zoom the image

**Comment by n-shevko** [comment]
Done. function extend_chain

https://github.com/Francesco215/requirements_for_self-organization_video/assets/11594789/bcad24fc-35ba-449e-9006-d6f24144be6a

**Comment by Francesco215** [comment]
Ok, that works, but can you make it smoother? (like un-zoomming at a constant speed)

**Comment by n-shevko** [comment]
I added rate_function=smooth, run_time=1.5 to shifting left. This made animation a bit smoother.

Another possible solution - make one animation from the following two:

scene.play(g.animate.scale(scale))
scene.play(g.animate.shift(LEFT * (new_radius + spacing)), rate_function=smooth, run_time=1.5)


https://github.com/Francesco215/requirements_for_self-organization_video/assets/11594789/00c87b45-c2b0-42c2-b883-1fa379434fae


But as I know animations on the same object can't be merged into one.

**Comment by Francesco215** [comment]
Have you tried using something like [`AnimationGroup`](https://docs.manim.community/en/stable/reference/manim.animation.composition.AnimationGroup.html)?

You might try to make the sequential drawing of the circles into one `AnimationGroup` and then run the animation group in parallel with a un-zooming animation `self.play(draw_cicles_anim_group,unzoom)`

I'm not 100% sure that this is the right way to go, but you can try and see what happens

**Comment by n-shevko** [comment]
fixed

https://github.com/Francesco215/requirements_for_self-organization_video/assets/11594789/9a88f4d9-4059-468a-91b2-410420605629

**Comment by Francesco215** [comment]
I made some changes to make it extra smooth

https://github.com/Francesco215/requirements_for_self-organization_video/assets/47751420/bfd0a3d1-db0c-4bfc-99d9-fe1b416f7ef5
